sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlatMapElementsTest.java

```diff
@@ -102,6 +105,51 @@ public void testFlatMapFnOutputTypeDescriptor() throws Exception {
     pipeline.run();
   }
 
+  /**
+   * A {@link SimpleFunction} to test that the coder registry can propagate coders
+   * that are bound to type variables.
+   */
+  private static class PolymorphicSimpleFunction<T> extends SimpleFunction<T, Iterable<T>> {
```

Comment:
We could also play with a few other simple functions:

SimpleFunction<T, T>
SimpleFunction<Iterable<T>, T> <- wouldn't expect this to work?